tools/checkissuecomment/checkissuecomment.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main contains linter for todo issue comments.
+package main
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/singlechecker"
+)
+
+var todoURLPrefix = "// TODO https://github.com/FerretDB/FerretDB/issues/"
+
+var analyzer = &analysis.Analyzer{
+	Name: "checkissuecomment",
+	Doc:  "check for TODO comments with issue links",
+	Run:  run,
+}
+
+func main() {
+	singlechecker.Main(analyzer)
+}
+
+// It analyses the presence of todo issue in code.
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		fileName := pass.Fset.File(file.Pos()).Name()
+		f, err := os.Open(fileName)
+		if err != nil {
+			return nil, err
+		}
+
+		defer func() {
+			err := f.Close()
+			if err != nil {
+				return
+			}
+		}()
+
+		scanner := bufio.NewScanner(f)
+		lineNumber := 1
+		var previousLine string
+
+		for scanner.Scan() {
+			line := scanner.Text()
+
+			if strings.Contains(line, todoURLPrefix) && strings.HasPrefix(previousLine, "//") {
+				pass.Reportf(file.Pos(), "TODO issue %s with comment %s found", line, previousLine)
+			}
```

Comment:
If I understand we want to allow following format:
```
// <comment>
// TODO https://github.com/FerretDB/FerretDB/issues/<number>
```
@AlekSi is that true?

Reply:
We could start with just `// TODO https://github.com/FerretDB/FerretDB/issues/<number>`

Reply:
so currently above PR supports only this pattern 
`// <comment>
// TODO https://github.com/FerretDB/FerretDB/issues/<number>`

Reply:
It seems to me that it returns error for such case. Can we start from something simple, like validating that every single line comment that starts with // TODO matches our pattern?

Reply:
ok so like just validate and print the lines matching pattern, but don't throw the in lint check error you mean?

Reply:
We need to ensure that issue links in comments look like this: `// TODO <url>`. I guess we can simply start by checking if there are no TODOs with issue number presented in any other way than that.
So if there's a comment like `// TODO 1234`, `// TODO #1234`, `// TODO https://github.com/FerretDB/FerretDB/1234 additional comment here` we want to return the message